Source/WebKit/UIProcess/AuxiliaryProcessProxy.cpp

```diff
@@ -591,10 +605,28 @@ void AuxiliaryProcessProxy::didChangeThrottleState(ProcessThrottleState state)
     if (m_isSuspended == isNowSuspended)
         return;
     m_isSuspended = isNowSuspended;
-#if ENABLE(CFPREFS_DIRECT_MODE)
-    if (!m_isSuspended && (!m_domainlessPreferencesUpdatedWhileSuspended.isEmpty() || !m_preferencesUpdatedWhileSuspended.isEmpty()))
-        send(Messages::AuxiliaryProcess::PreferencesDidUpdate(std::exchange(m_domainlessPreferencesUpdatedWhileSuspended, { }), std::exchange(m_preferencesUpdatedWhileSuspended, { })), 0);
-#endif
+
+    if (!isNowSuspended && !m_messagesToSendOnResume.isEmpty()) {
+        Vector<std::pair<unsigned, std::unique_ptr<IPC::Encoder>>> indexMessagePairs;
+        indexMessagePairs.reserveInitialCapacity(m_messagesToSendOnResume.size());
+
+        for (auto& indexMessagePair : m_messagesToSendOnResume.values())
+            indexMessagePairs.append(WTFMove(indexMessagePair));
```

Comment:
Should we set a limit on number of messages to send when resuming a process? Is the coalescing enough to prevent us from spamming the newly-unsuspended process with messages such that it hangs on resume?

Reply:
I don't think it's a particular concern for now. We already spam a resuming process with up to 256 messages at once (see `largeOutgoingMessageQueueCountThreshold`--it's used to size two different queues) and given the coalescing here I think we'll be doing much less spamming from this codepath. But if necessary we can add a similar limit here in the future.